### PR TITLE
dcmtk: fix compilation error `ambiguous overload for operator==` with some compilers

### PR DESCRIPTION
**Pull request by SpaceIm**
closes https://github.com/conan-io/conan-center-index/issues/20332
closes https://github.com/conan-io/conan-center-index/issues/20444

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/dcmtk/all/conandata.yml [hunk 1]
```diff
@@ -6,7 +6,10 @@ patches:
   "3.6.7":
     - patch_file: "patches/3.6.7-0001-cmake-robust-deps-handling.patch"
       patch_description: "CMake: robust discovery with find_package() and use imported targets"
-      patch_type: conan
+      patch_type: "conan"
     - patch_file: "patches/3.6.7-0002-cmake-check-openssl-symbol.patch"
       patch_description: "CMake: fix OpenSSL compatibility checks"
-      patch_type: conan
+      patch_type: "conan"
+    - patch_file: "patches/3.6.7-0003-ambiguous-overload-operator-equal.patch"
```
Comment (RubenRBS) [inline recipes/dcmtk/all/conandata.yml:13]: Hi! Thanks a lot for your contribution - As this is a C++ change, it needs a bit more consideration. Has this been reported upstream? I'm guessing no as theres no patch_source field.

After checking the code changes they look good to me, but would still appreciate an lib author confirmation on their side if you could get one :)

**Review by RubenRBS — COMMENTED** [review]
Thanks a lot! As this is a C++ code change (Which looks good to me!) as per our policy, we'd like to have some extra checks before acceping them

File: recipes/dcmtk/all/conandata.yml (hunk 1 above)
Comment (SpaceIm) [reply recipes/dcmtk/all/conandata.yml:13]: I'll do when I have time. There is no way to report issues in dcmtk repo, only PR.

Comment (RubenRBS) [reply recipes/dcmtk/all/conandata.yml:13]: Thanks a lot :) Note that I've checked and the changes look valid to me, but I'd like to still have a second opinion, in unlikely case that it somehow is not correct :)

Thanks a lot for your work

Comment (SpaceIm) [reply recipes/dcmtk/all/conandata.yml:13]: yep, I'll submit a PR upstream today or tomorrow.

Comment (SpaceIm) [reply recipes/dcmtk/all/conandata.yml:13]: I've submitted https://github.com/DCMTK/dcmtk/pull/88

Comment (SpaceIm) [reply recipes/dcmtk/all/conandata.yml:13]: Turn to draft, it might be fixed by defining other operator==

Comment (SpaceIm) [reply recipes/dcmtk/all/conandata.yml:13]: Upstream seems to agree with the new patch I've submitted (define additional `operator==`): https://github.com/DCMTK/dcmtk/pull/88#issuecomment-1760322478

**Comment by RubenRBS** [comment]
@SpaceIm do you plan to fix the hook system lib warning?

**Comment by SpaceIm** [comment]
No because I think that it's not portable to all Linux distributions.

**Review by jwillikers — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Review by valgur — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**